--- rtdetrv2_pytorch/tools/export_onnx.py	2025-09-26 20:42:56.942999592 -0700
+++ rtdetrv2_pytorch_fix/tools/export_onnx.py	2025-09-26 20:43:58.113036177 -0700
@@ -59,7 +59,7 @@
         input_names=['images', 'orig_target_sizes'],
         output_names=['labels', 'boxes', 'scores'],
         dynamic_axes=dynamic_axes,
-        opset_version=16, 
+        opset_version=args.opset, 
         verbose=False,
         do_constant_folding=True,
     )
@@ -90,7 +90,7 @@
     parser.add_argument('--input_size', '-s', type=int, default=640)
     parser.add_argument('--check',  action='store_true', default=False,)
     parser.add_argument('--simplify',  action='store_true', default=False,)
-    
+    parser.add_argument('--opset', type=int, default=17)
 
     args = parser.parse_args()
 
