--- export_onnx.py	2026-02-04 13:47:36.471562151 -0600
+++ export_onnx_modified.py	2026-02-04 13:48:07.942685488 -0600
@@ -54,6 +54,13 @@
         action="store_true",
         help="decode in inference or not"
     )
+    parser.add_argument(
+        "--imgsz",
+        default=None,
+        type=int,
+        nargs=2,
+        help="override test size (height width)",
+    )
 
     return parser
 
@@ -65,6 +72,9 @@
     exp = get_exp(args.exp_file, args.name)
     exp.merge(args.opts)
 
+    if args.imgsz is not None:
+        exp.test_size = tuple(args.imgsz)
+
     if not args.experiment_name:
         args.experiment_name = exp.exp_name
 
@@ -88,7 +98,7 @@
     logger.info("loading checkpoint done.")
     dummy_input = torch.randn(args.batch_size, 3, exp.test_size[0], exp.test_size[1])
 
-    torch.onnx._export(
+    torch.onnx.export(
         model,
         dummy_input,
         args.output_name,
@@ -102,15 +112,13 @@
 
     if not args.no_onnxsim:
         import onnx
-        from onnxsim import simplify
+        import onnxslim
 
         # use onnx-simplifier to reduce reduent model.
         onnx_model = onnx.load(args.output_name)
-        model_simp, check = simplify(onnx_model)
-        assert check, "Simplified ONNX model could not be validated"
+        model_simp = onnxslim.slim(onnx_model)
         onnx.save(model_simp, args.output_name)
         logger.info("generated simplified onnx model named {}".format(args.output_name))
 
-
 if __name__ == "__main__":
     main()
